front-end/src/pages/SearchEvents/SearchEvents.jsx

```diff
@@ -7,22 +8,80 @@ import Form from 'react-bootstrap/Form';
 import InputGroup from 'react-bootstrap/InputGroup';
 import './SearchEvents.css'; // Assuming you add a CSS file for extra styles
 import { getEventDetailsSearch } from "../../utilities/EventUtilities";
+import { getUserProfile } from '../../utilities/UserProfileUtilities'
 import EventCard from "../../components/EventCard";
+import { getCountries, getStates, getCities } from "../../utilities/CountryStateCityUtilities";
 
 function SearchEvents() {
+    const [userLocations, setUserLocations] = useState([]);
     const [searchType, setSearchType] = useState('');
     const [searchTerm, setSearchTerm] = useState('');
     const [searchDateStart, setSearchDateStart] = useState('');
     const [searchDateEnd, setSearchDateEnd] = useState('');
     const [searchLocation, setSearchLocation] = useState('');
+    const [locationData, setLocationData] = useState([]);
     const [searchEventType, setSearchEventType] = useState('');
     const [searchEvents, setSearchEvents] = useState([]);
     const [eventsPopular, setEventsPopular] = useState([]);
     // TODO: once we have a spot on our events to indicate whether volunteers are needed, we can add functionality to sort searchEvents into searchEventsVolNeed
     const [eventsVolNeed, setEventsVolNeed] = useState([]);
     const [eventsAdditional, setEventsAdditional] = useState([]);
-    console.log('searchEvents', searchEvents)
+    // Next three set api data for auto-populated suggestions
+    const [apiCountries, setApiCountries] = useState([]);
+    const [apiStates, setApiStates] = useState([]);
+    const [apiCities, setApiCities] = useState([]);
+    // Next three set location data from form to be used in formatting and setting the userLocation
+    const [countryAdd, setCountryAdd] = useState("");
+    const [stateAdd, setStateAdd] = useState("");
+    const [cityAdd, setCityAdd] = useState("");
+    const myOutletContextObj = useOutletContext();
+    const { user } = myOutletContextObj;
```

Comment:
just curious - why did you set it up like this instead of just 
`const { user } = useOutletContext()` ?